gpt_engineer/steps.py

```diff
@@ -2,141 +2,167 @@
 import re
 import subprocess
 
-from gpt_engineer.ai import AI
+from typing import Any, Callable, Dict, List, Tuple
+
+from gpt_engineer.ai.ai import AI
 from gpt_engineer.chat_to_files import to_files
 from gpt_engineer.db import DBs
+from gpt_engineer.models import Message, Role, Step
 
 
-def setup_sys_prompt(dbs):
+def setup_sys_prompt(dbs: DBs) -> str:
+    """Constructs the system setup prompt."""
     return dbs.identity["generate"] + "\nUseful to know:\n" + dbs.identity["philosophy"]
 
 
-def simple_gen(ai: AI, dbs: DBs):
-    """Run the AI on the main prompt and save the results"""
-    messages = ai.start(
-        setup_sys_prompt(dbs),
-        dbs.input["main_prompt"],
-    )
-    to_files(messages[-1]["content"], dbs.workspace)
-    return messages
+def parse_message_json(x: Dict[str, Any]) -> Tuple[Role, Message]:
+    return Role(x["role"]), Message(x["content"])
+
 
+### Steps
 
-def clarify(ai: AI, dbs: DBs):
+
+def clarify(ai: AI, dbs: DBs) -> List[Tuple[Role, Message]]:
     """
-    Ask the user if they want to clarify anything and save the results to the workspace
+    Ask the user if they want to clarify anything and save the results to the workspace.
     """
-    messages = [ai.fsystem(dbs.identity["qa"])]
-    user = dbs.input["main_prompt"]
+    user_msg = dbs.input["main_prompt"]
```

Comment:
Sorry, could you elaborate on this, I don't understand. 
```python
user_msg = dbs.input["main_prompt"]
messages = [(Role.SYSTEM, Message(dbs.identity["qa"]))]
```
Is needed before the while loop because updates are not equivalent.